sources/osgUtil/Composer.js

```diff
@@ -355,48 +426,172 @@ define( [
-                            var uniformValue = this._uniforms[ uniformName ];
-                            if ( uniformType.search( 'sampler' ) !== -1 ) {
-                                this._stateSet.setTextureAttributeAndModes( unitIndex, uniformValue );
-                                uniform = Uniform.createInt1( unitIndex, uniformName );
-                                unitIndex++;
-                                this._stateSet.addUniform( uniform );
-                            } else {
-                                if ( Uniform.isUniform( uniformValue ) ) {
-                                    uniform = uniformValue;
-                                } else {
-                                    uniform = Uniform[ uniformType ]( this._uniforms[ uniformName ], uniformName );
-                                }
-                                this._stateSet.addUniform( uniform );
-                            }
-                        }
-                    }
-                }
+                this.autoBindFragmentUniformStateSet( this._stateSet, this._fragmentShader, this._uniforms );
             }
-            this._stateSet.setAttributeAndModes( program );
+            this._stateSet.setAttributeAndModes( this._program );
             this._dirty = false;
         }
     } );
 
 
+    // filter that switch its render target and its input at each frame
+    // allowing to get input for last frame render.
+    Composer.Filter.PingPong = function ( cameraRtt0, rtt0, cameraRtt1, rtt1, fragmentShader, uniforms ) {
+        Composer.Filter.Custom.apply( this, [ fragmentShader, uniforms ] );
+
+        this._cameraRtt0 = cameraRtt0;
+        this._rtt0 = rtt0;
+
+        this._cameraRtt1 = cameraRtt1;
+        this._rtt1 = rtt1;
+
+        this._fragmentName = 'PingPong';
+    };
+
+    Composer.Filter.PingPong.prototype = MACROUTILS.objectInherit( Composer.Filter.Custom.prototype, {
+
+        // Constraints:
+        // - Next Filter: texture unit 0 === rtt0 && texture unit 1 === rtt1
+        // - Previous Filter: the output of the previous filter if any
+        //      will be binded to texture unit 0 of both camera stateset
+        interConnectFilters: function ( composer, i, array ) {
+
+            var filterStateSet = this.getStateSet();
+
+            var st0 = this._cameraRtt0.getOrCreateStateSet();
+            var st1 = this._cameraRtt1.getOrCreateStateSet();
+
+            // copy filter program and uniforms on the 2 cameras
+            st0.setAttributeAndModes( this._program );
+            st1.setAttributeAndModes( this._program );
+
+            if ( this._uniforms ) {
+                for ( var k = 0; k < this._uniforms.length; k++ ) {
+                    st0.addUniform( this._uniforms[ k ] );
+                    st1.addUniform( this._uniforms[ k ] );
+                }
+            }
+
+            st0.addUniform( Uniform.createInt1( 0, 'Texture0' ) );
```

Comment:
you can use the same uniform for st0 and st1. no need to create Texture0 and Texture1 twice